tests.md

```diff
@@ -0,0 +1,174 @@
+# Testing of Terraform Modules Explained
+
+## Are Unit Tests Viable?
+
+A useful definition of a unit test is one which mainly spends its run time in the code that we own, and very little
+time in the code owned by other organizations or teams. There are other definitions, but the usefulness of this
+is dictated by tightening the loop of developer's work. The progress is faster with a five-seconds test run than with
+a five-hours test run as the developer waits for the result to decide whether to go further with the work or to go
+back and correct the last portion of their work. The more time the test run uses for the others' code, the less time
+it spends inside our code, to catch the kind of errors *this team* is able to fix by ourselves.
```

Comment:
you may want to scope the unit tests down to the method/function level. 

A five hour test is still valuable from the stand point of how that method interacts with the rest of the code base, and should run less often.

Reply:
Yes. (Does the paragraph need corrections/modification/deletion?)